src/app/ClusterInfo.h

```diff
@@ -79,6 +79,11 @@ struct ClusterInfo
     // For event, an event id can only be interpreted if the cluster id is known.
     bool IsValidEventPath() const { return !(HasWildcardClusterId() && !HasWildcardEventId()); }
 
+    bool IsValidDataVersionFilter() const
+    {
+        return !HasWildcardEndpointId() && !HasWildcardClusterId() && mDataVersion != kUndefinedDataVersion;
```

Comment:
Again, there is no such thing as kUndefinedDataVersion.  0 is a valid value.

Reply:
fixed, use optional instead.